### DISCUSSION TITLE
What does a "good" logical flow look like with Frigate, HA, and frigate lovelace card?  (RTSP?)

### DISCUSSION THREAD

**Discussion by rhatguy** [blakeblackshear/frigate#1790] [General]
Not sure if this is the right place to ask this, but the awesome frigate lovelace card by @dermotduffy is able to leverage WebRTC to provide efficient realtime view, however there doesn't seem to be a way to pull RTSP streams from frigate.  I realize there are rtsp proxies that could be implemented in front of frigate and I believe @blakeblackshear has mentioned the possibility of frigate implementing some type of proxy functionality in the future also.

Just looking for some guidance as to where the puck is moving on these projects.  What are the authors future visions for what a "good" logical flow looks like with Frigate, Home Assistant, and Frigate Lovelace card that would ideally limit the number of connections to the cameras and provide the best "live" views of what frigate is ultimately seeing through Home Assistant?

**Comment by blakeblackshear** [comment]
The goal is to enable real-time, low latency viewing of feeds in lovelace with the card using the same technology (jsmpeg) used in frigate's frontend for the 0.9.0 release. In the end state, you should be able to install the integration and use the lovelace card without the need for webrtc.

With that said, I do think webrtc is a good alternative and should be supported too. I do plan to implement a more robust relay for video steams in frigate that supports rtsp and hls in addition to rtmp.

**Comment by dermotduffy** [comment]
There are two parts of getting JSMPEG working (proxy JSMPEG stream through the integration, allow the card to render it). Both are already merged, but neither is part of a release -- won't be too long. JSMPEG is a bit of  a compromise option -- it's quite a bit lower latency [better] than than the RTMP (via HA) stream, but [anecdotally] with a lower framerate / lesser experience than WebRTC  can offer.

I do think native WebRTC is something Frigate would ideally support. Just proxying the RTSP stream @ a predictable URL would simplify the setup for the card quite a bit.
